keyboards/sofle/keymaps/nartudy/keymap.c

```diff
@@ -0,0 +1,784 @@
+             // oled_write_raw_P(prep[abs((PREP_FRAMES-1)-current_prep_frame)], ANIM_SIZE); // uncomment if IDLE_FRAMES >1
+             oled_write_raw_P(prep[0], ANIM_SIZE);  // remove if IDLE_FRAMES >1
+         }
+         if(get_current_wpm() >=TAP_SPEED){
+             current_tap_frame = (current_tap_frame + 1) % TAP_FRAMES;
+             oled_write_raw_P(tap[abs((TAP_FRAMES-1)-current_tap_frame)], ANIM_SIZE);
+         }
+    }
+    if(get_current_wpm() != 000) {
+        oled_on(); // not essential but turns on animation OLED with any alpha keypress
+        if(timer_elapsed32(anim_timer) > ANIM_FRAME_DURATION) {
+            anim_timer = timer_read32();
+            animation_phase();
+        }
+        anim_sleep = timer_read32();
+    } else {
+        if(timer_elapsed32(anim_sleep) > OLED_TIMEOUT) {
+            oled_off();
+        } else {
+            if(timer_elapsed32(anim_timer) > ANIM_FRAME_DURATION) {
+                anim_timer = timer_read32();
+                animation_phase();
+            }
+        }
+    }
+}
+
+ static void print_status_narrow(void) {
+    /* Print current mode */
+    oled_set_cursor(0,0);
+    if (keymap_config.swap_lctl_lgui) {
+        oled_write_raw_P(mac_logo, sizeof(mac_logo));
+    } else {
+        oled_write_raw_P(windows_logo, sizeof(windows_logo));
+    }
+
+    oled_set_cursor(0,3);
+
+    switch (get_highest_layer(default_layer_state)) {
+        case _QWERTY:
+            oled_write("QWRT", false);
+            break;
+        case _COLEMAK:
+            oled_write("WASD", false);
+            break;
+        default:
+            oled_write("UNDEF", false);
+    }
+
+    oled_set_cursor(0,5);
+
+    /* Print current layer */
+    oled_write("LAYER", false);
+
+    oled_set_cursor(0,6);
+
+    switch (get_highest_layer(layer_state)) {
+        case _QWERTY:
+            oled_write("Base ", false);
+            break;
+        case _COLEMAK:
+            oled_write("GAME/NUM", false);
+            break;
+        case _RAISE:
+            oled_write("Raise", false);
+            break;
+        case _LOWER:
+            oled_write("Lower", false);
+            break;
+        case _ADJUST:
+            oled_write("Adj  ", false);
+            break;
+        default:
+            oled_write("Undef", false);
+    }
+
+    /* caps lock */
+    oled_set_cursor(0,8);
+    oled_write("CPSLK", led_usb_state.caps_lock);
+
+    /* KEYBOARD PET RENDER START */
+
+    render_luna(0,13);
+
+    /* KEYBOARD PET RENDER END */
+}
+void oled_task_user(void) {
+
+    /* KEYBOARD PET VARIABLES START */
+
+    current_wpm = get_current_wpm();
+    led_usb_state = host_keyboard_led_state();
+
+    /* KEYBOARD PET VARIABLES END */
+
+    if (is_keyboard_master()) {
+          print_status_narrow();
+
+      
+    } else {
+        render_anim();
+        oled_set_cursor(0,6);
+        sprintf(wpm_str, "WPM: %hu", get_current_wpm());
+        oled_write(wpm_str, false);
```

Comment:
This reduces the firmware by about 1.5K. 
```suggestion
        char wpm_counter[4];
        wpm_counter[3] = '\0';
        wpm_counter[2] = '0' + n % 10;
        wpm_counter[1] = (n /= 10) % 10 ? '0' + (n) % 10 : (n / 10) % 10 ? '0' : ' ';
        wpm_counter[0] = n / 10 ? '0' + n / 10 : ' ';
        oled_write_P(PSTR("WPM: "), false);
        oled_write(wpm_counter, false);
```